clang/lib/AST/DeclCXX.cpp

```diff
@@ -2148,6 +2158,19 @@ void CXXRecordDecl::completeDefinition(CXXFinalOverriderMap *FinalOverriders) {
   for (conversion_iterator I = conversion_begin(), E = conversion_end();
        I != E; ++I)
     I.setAccess((*I)->getAccess());
+
+  ASTContext &Context = getASTContext();
+  if (!Context.getLangOpts().CPlusPlus20 && hasUserDeclaredConstructor()) {
```

Comment:
We should probably complaint only if the type is aggregate in the first place, right?
See https://gcc.godbolt.org/z/sa94xa9or for an example code.